src/huak/project/mod.rs

```diff
@@ -6,6 +6,16 @@ use crate::errors::HuakError;
 
 use self::config::Config;
 
+/// There are two kinds of project, application and library.
+/// Application projects usually have one or more entrypoints in the form of
+/// runnable scripts while library projects do not.
+#[derive(Default)]
+pub enum ProjectType {
+    #[default]
+    Application,
+    Library,
+}
```

Comment:
I'm wondering if we should default to Library. What do you think? Especially if the Application comes with a baked console script.

Reply:
yeah i went back and forth a little bit on that. i ended up on app to mirror cargo's default of bin, but given that it adds extra functionality i definitely see the appeal of defaulting to the simpler lib option.

Reply:
defaulted to library